src/trs/hmtrs.rs

```diff
@@ -246,71 +285,45 @@ impl HMTRS {
             .into_iter()
             .filter(|(o, _)| o == &Some(term.head()))
             .collect();
-        // fail if not exactly 1 match
```

Comment:
One thing I really disliked about my initial implementation, and which this code inherits from it, is that we should immediately fail if there's more than one match. That's a structural constraint on the setup of the lexicon which the code doesn't capture well. Any ideas for resolving that?